### PR TITLE
Add a doc page for Presto Console development

### PR DESCRIPTION
**Pull request by yhwang**
## Description
Cover the basic information for the Presto Console development, including
- where the Presto Console code is located
- code convention
- how to add a new page

Use this page to hold the necessary information for Presto Console development. More information will be added later.

related: #21062 item 9

## Motivation and Context
Add the doc to cover the basic knowledge about Presto Console development.

## Impact
A new doc page under the `Developer Guide` section

## Test Plan
Generate the doc pages and verify the new content.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-docs/src/main/sphinx/develop/web-ui.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+- ``assets`` folder: contains Presto favicon, logo, and Presto CSS file.
+- ``dev`` directory: contains ``Query Viewer`` page and its JS file.
+- ``dist`` directory: contains the JS files generated by webpack.
+- ``vendor`` directory: contains third-party JS/CSS libraries that are used by UI components.
+- ``src`` directory: contains the UI components and webpack config file.
+
+Prerequisites
+-------------
+
+- `yarn <https://yarnpkg.com/>`_: A package manager for the Node.js project. It's used to manage the packages
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/develop/web-ui.rst:27]:
```suggestion
- `yarn <https://yarnpkg.com/>`_: A package manager for the Node.js project. Yarn is used to manage the packages
```

File: presto-docs/src/main/sphinx/develop/web-ui.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+-------------
+
+All of the UI components are in the ``src`` folder under the ``web UI root`` directory.  These components are
+written in `JSX <https://react.dev/learn/writing-markup-with-jsx>`_ using React JS and named with the ``.jsx``
+file extension.
+
+Please follow these rules for composing UI components:
+
+- Define components as JavaScript functions instead of classes: Some old components are written in classes and
+  will be refactored into functions. For new components, please use function components. You can check
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/develop/web-ui.rst:40]:
```suggestion
  should be refactored into functions. For new components, please use function components. Check
```
should be? must be? what do you think fits better?

File: presto-docs/src/main/sphinx/develop/web-ui.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  `here <https://react.dev/learn/your-first-component#defining-a-component>`_ for detailed information.
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/develop/web-ui.rst:41]:
```suggestion
  `Defining a component <https://react.dev/learn/your-first-component#defining-a-component>`_ for detailed information.
```
See Gitlab doc style guide [Text for links](https://docs.gitlab.com/ee/development/documentation/styleguide/#text-for-links) for context.

File: presto-docs/src/main/sphinx/develop/web-ui.rst [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+- Bootstrap: the current code base is using Bootstrap v3.3.5. You can find the CSS and components it provides
+  `here <https://bootstrapdocs.com/v3.3.5/docs/getting-started/>`_.
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/develop/web-ui.rst:43]:
```suggestion
  in the `Bootstrap Getting Started <https://bootstrapdocs.com/v3.3.5/docs/getting-started/>`_.
```

File: presto-docs/src/main/sphinx/develop/web-ui.rst [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+- Use `flow <https://flow.org/>`_ as the static type checker:  Add ``//@flow`` at the beginning of the ``.jsx`` file
+  to enable the type checker and use ``yarn run flow`` to run the flow checker.
+
+Web Dev Server
+--------------
+
+To speed up the UI development, you can use the web dev server that webpack provides to verify new UI components or updates
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/develop/web-ui.rst:50]:
```suggestion
To speed up the UI development, use the web dev server that webpack provides to verify new UI components or updates
```

File: presto-docs/src/main/sphinx/develop/web-ui.rst [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  Pay attention to the output messages and look for messages like ``[webpack-dev-server] Loopback: http://localhost:8081/``.
+  It tells you where to access the web dev server.
+
+The web dev server watches the entries listed in ``webpack.config.js``, compiles, and serves the HTML pages in the ``web UI root``
+directory.
+
+Add a New Page
+--------------
+
+To add a new page, you need to:
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/develop/web-ui.rst:67]:
```suggestion
To add a new page:
```

File: presto-docs/src/main/sphinx/develop/web-ui.rst [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+1. Add a new HTML page to the ``web UI root`` directory. Inside the HTML page, add the necessary ``<link>`` tags to include
+   the needed CSS and JavaScript libraries from the ``vendor`` directory or external public CDN URIs. Inside the ``<body>`` element,
+   add an ``<div>`` element with the ``id`` attribute as the placeholder for the UI component you are going to create
+   followed by a ``<script>`` tag to point to the new JavaScript file that would be generated by the webpack at the ``dist`` directory.
+   Check ``query.html`` as an example.
+2. Add a new ``.jsx`` file under the ``src`` directory and a corresponding entry in the ``webpack.config.js``. This ``.jsx`` file would be
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/develop/web-ui.rst:74]:
```suggestion
2. Add a new ``.jsx`` file under the ``src`` directory and a corresponding entry in the ``webpack.config.js``. This ``.jsx`` file is
```

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for the doc! A few nits and small suggestions, I really like the examples and detail on how to do this.

**Comment by yhwang** [comment]
@steveburnett 
> I want to discuss what we call the Web UI

I kept thinking about the term `Web UI` while writing this doc. Now you remind me of the `Presto Console`. Let me change it to `Presto Console`. Thanks!

File: presto-docs/src/main/sphinx/develop/web-ui.rst [hunk 8]
```diff
@@ -0,0 +1,81 @@
+======
+Web UI
+======
+
+Presto comes with a built-in dashboard where you can check the cluster’s information, running SQL queries,
+query details, resource group information, etc. Presto web UI comprises a series of HTML pages and JavaScript
```
[the remark this answers is not in the document]
Comment (yhwang) [reply presto-docs/src/main/sphinx/develop/web-ui.rst:6]: I guess I should bookmark this page for reference and revisit it whenever I work on documentation. :-)

File: presto-docs/src/main/sphinx/develop/web-ui.rst (hunk 2 above)
Comment (yhwang) [reply presto-docs/src/main/sphinx/develop/web-ui.rst:40]: vote for `must be`.

**Comment by yhwang** [comment]
@steveburnett I updated the PR to use `Presto Console` and incorporated all your comments (I believe :-p).
Thanks for the review and comments.

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

Thanks for the quick revision!

**Pull request merged**